native/ex_tokenizers/src/models.rs

```diff
@@ -362,23 +363,34 @@ pub fn models_wordlevel_from_file(
 #[derive(NifTaggedEnum)]
 pub enum UnigramOption {
     UnkId(usize),
+    ByteFallback(bool),
 }
 
 #[rustler::nif]
 pub fn models_unigram_init(
     vocab: Vec<(String, f64)>,
     options: Vec<UnigramOption>,
 ) -> Result<ExTokenizersModel, ExTokenizersError> {
-    let unk_id = if !options.is_empty() {
-        match options[0] {
-            UnigramOption::UnkId(unk_id) => Some(unk_id),
-        }
-    } else {
-        None
+    let unk_id = match options
+        .iter()
+        .find(|opt| matches!(opt, UnigramOption::UnkId(_)))
+        .unwrap()
+    {
+        UnigramOption::UnkId(unk_id) => Some(*unk_id),
+        _ => None,
+    };
+
+    let byte_fallback = match options
```

Comment:
Are we matching  the same 2 times (here and inside `matches!`)? Is it compiler optimised?

Reply:
Nah, the outer match is on the found inner match, which is on each element.